module/vision/VisualMesh/src/VisualMesh.cpp

```diff
@@ -0,0 +1,174 @@
+                    lens.focal_length = 1.0 / cam.radial.radiansPerPixel;
+                    break;
+                case CameraParameters::LensType::PINHOLE:
+                    lens.projection   = mesh::VisualMesh<float>::Lens::RECTILINEAR;
+                    lens.focal_length = cam.pinhole.focalLengthPixels;
+                    break;
+            }
+            lens.dimensions = {{int(img.dimensions[0]), int(img.dimensions[1])}};
+            lens.fov        = cam.FOV.x();
+
+            auto results = classifier(img.data.data(), mesh::VisualMesh<float>::FOURCC(img.format), Hoc, lens);
+
+            // Get the mesh that was used so we can make our message
+            const auto& m = mesh.height(Hoc[2][3]);
+
+            auto msg = std::make_unique<VisualMeshMsg>();
+
+            // Pass through our camera id
+            msg->camera_id = img.camera_id;
+
+            // Add our description
+            for (const auto& r : m.rows) {
+                msg->mesh.emplace_back(r.phi, r.end - r.begin);
+            }
+
+            // Add our indices
+            msg->indices = std::move(results.global_indices);
+
+            // Add our neighbourhood
+            msg->neighbourhood.reserve(results.neighbourhood.size());
+            for (const auto& n : results.neighbourhood) {
+                msg->neighbourhood.emplace_back(Eigen::Map<const Eigen::Matrix<int, 6, 1, Eigen::DontAlign>>(n.data()));
+            }
+
+            // Add our coordinates
+            msg->coordinates = results.pixel_coordinates.as<Eigen::Matrix<int, 2, 1, Eigen::DontAlign>>();
+
+            // Add our classifications
+            // Add our first (the image) and last (the results) to our list
+            for (const auto& c : results.classifications) {
+                msg->classifications.emplace_back(c.first, c.second);
+            }
+
+
+            // -- Graphing for NUsight
+            msg->classifications.emplace_back(results.classifications.front().first,
+                                              results.classifications.front().second);
+            msg->classifications.emplace_back(results.classifications.back().first,
+                                              results.classifications.back().second);
+
+
+            std::vector<std::tuple<Eigen::Vector2i, Eigen::Vector2i, Eigen::Vector4d>,
+                        Eigen::aligned_allocator<std::tuple<Eigen::Vector2i, Eigen::Vector2i, Eigen::Vector4d>>>
+                lines;
+
+
+            std::vector<std::array<int, 2>> pixel_coordinates;
+            std::vector<std::array<float, 2>> lazyConvert = results.pixel_coordinates;
+            pixel_coordinates.reserve(lazyConvert.size());
+            for (int i = 0; i < lazyConvert.size(); ++i) {
+                pixel_coordinates[i] = {int(lazyConvert[i][0]), int(lazyConvert[i][1])};
+            }
+
+            std::vector<float> classification = results.classifications.front().second;
```

Comment:
Later you might want to change this to back to see the classifications rather than the image itself (also change below to output classes rather than colours)